platform/lib/pipeline/markdownDocument.js

```diff
@@ -334,6 +334,32 @@ class MarkdownDocument {
     return true;
   }
 
+  /**
+   *Adds explicit anchors for titels in github notation
+   *
+   */
+  addExplicitAnchors() {
+    const TITLE_PATTERN = /^#+[ \t]+(?!.*<a[ \t]+name)(.+?)[ \t]*$/mg;
+    const existingSlugs = [];
+    this._contents = this._contents.replace(TITLE_PATTERN, (line, headline) => {
+      let slug = headline.toLowerCase();
```

Comment:
please extract the slug creation logic into a helper class (ideally outside markdown document).